lasagne/layers/base.py

```diff
@@ -189,28 +190,33 @@ def create_param(param, shape):
                 its output is used to initialize the variable.
 
         :note:
-            This static method should be used in `__init__()` when creating a
+            This method should be used in `__init__()` when creating a
             :class:`Layer` subclass that has trainable parameters. This
             enables the layer to support initialization with numpy arrays,
             existing Theano shared variables, and callables for generating
             initial parameter values.
         """
-        if isinstance(param, np.ndarray):
-            if param.shape != shape:
-                raise RuntimeError("parameter array has shape %s, should be %s" % (param.shape, shape))
-            return theano.shared(param)
+        if name is not None:
+            if self.name is not None:
+                name = "%s.%s" % (self.name, name)
```

Comment:
If this isn't a static method any more, I need to change back some of the tests in #52. This will remind me of it.